### PR TITLE
Add sync scroll button

### LINKED ISSUE #478: Disable sync scroll button
**Is your feature request related to a problem? Please describe.**
In codimd 1.6 you could disable the synced scroll.

**Describe the solution you'd like**
Add a toggle to the document bar, that enables or disables the sync scroll.

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Editor

### Description
This PR adds a "disable/enable sync scroll" button.

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
Closes #478

### CODE REVIEW

File: src/components/editor/editor.tsx [hunk 1]
```diff
@@ -68,21 +70,23 @@ export const Editor: React.FC = () => {
...
-    if (scrollSource.current === ScrollSource.RENDERER) {
+    debugger
+    if (scrollSource.current === ScrollSource.RENDERER && editorSyncScroll) {
       setScrollState((old) => ({ editorScrollState: newScrollState, rendererScrollState: old.rendererScrollState }))
     }
-  }, [])
+  }, [editorSyncScroll])
+
+  const onEditorScroll = useCallback((newScrollState: ScrollState) => {
+    debugger
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply src/components/editor/editor.tsx]: removed

File: src/components/editor/document-bar/buttons/sync-scroll-button.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { useCallback } from 'react'
+import { ToggleButton, ToggleButtonGroup } from 'react-bootstrap'
+import { Trans } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import { ApplicationState } from '../../../../redux'
+import { setEditorSyncScroll } from '../../../../redux/editor/methods'
+import { ForkAwesomeIcon } from '../../../common/fork-awesome/fork-awesome-icon'
+
+export const SyncScrollButton: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply src/components/editor/document-bar/buttons/sync-scroll-button.tsx]: added

File: src/components/editor/document-bar/buttons/sync-scroll-button.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const syncScroll: boolean = useSelector((state: ApplicationState) => state.editorConfig.syncScroll)
+  const translation = syncScroll ? 'editor.documentBar.disableSyncScroll' : 'editor.documentBar.enableSyncScroll'
+  const iconColor = syncScroll ? '' : 'text-muted'
+  const onClick = useCallback(() => {
+    setEditorSyncScroll(!syncScroll)
+  }, [syncScroll])
+
+  return (
+    <ToggleButtonGroup type="checkbox" defaultValue={[]} name="dark-mode" className="ml-2" value={[syncScroll]}>
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply src/components/editor/document-bar/buttons/sync-scroll-button.tsx]: changed

**Review by ErikMichelson — APPROVED** [review]

**Review by DerMolly — COMMENTED** [review]
I think the logo should be changed to something like these
grafik. I'd also suggest to leave the sync arrows and just have one of those in a normal and in a strikethrough version.

**Comment by mrdrogdrog** [comment]
image
image

**Review by DerMolly — APPROVED** [review]

**Pull request merged**